lib/Service/AccountService.php

```diff
@@ -181,4 +181,21 @@ public function updateSignature(int $id, string $uid, string $signature = null):
 	public function getAllAcounts(): array {
 		return $this->mapper->getAllAccounts();
 	}
+
+	
+	/**
+	 * @param string $currentUserId
+	 * @param int $accountId
+	 * @return bool
+	 */
+
+	public function testAccountConnection(string $currentUserId, int $accountId) :bool {
+		$account = $this->find($currentUserId, $accountId);
+		try {
+			$account->getImapConnection();
```

Comment:
``\OCA\Mail\Account::getImapConnection`` is deprecated and should not be used anymore.

@hamza221 please refactor the code to use the IMAP client factory and add a unit test for this method.

Reply:
#8962